### ISSUE TITLE
🐛 BUG: `with-tailwindcss` example: Button component is unstyled on Markdown page

### ISSUE THREAD

**Issue by phenomen** [withastro/astro#2995]
### What version of `astro` are you using?

1.0.0-beta.0

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

1. Open [with-tailwind](https://github.com/withastro/astro/tree/main/examples/with-tailwindcss) example
2. Click `Markdown is also supported...`

Expected: Button component has the same style as on the non-markdown page
Actual: Button component is unstyled

### Link to Minimal Reproducible Example

https://codesandbox.io/s/github/withastro/astro/tree/latest/examples/with-tailwindcss

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by tony-sull** [comment]
Oddly this doesn't hit for me locally but does hit in Code Sandbox 🤔 

@phenomen Does adding `md` to the list of file extensions in the tailwind config file do the trick?  We may just need to add that so Tailwind's JIT compiler tracks markdown files as well

```
module.exports = {
	content: ['./src/**/*.{astro,html,md,js,jsx,svelte,ts,tsx,vue}'],
	theme: {
		extend: {},
	},
	plugins: [],
};
```

**Comment by tony-sull** [comment]
Not a config issue after all! Still not sure why this doesn't reproduce for me locally, but the `<Button>` component used in the example markdown is actually hit a case sensitivity bug somewhere, it's being rendered as a plain old `<button>` element

Renaming the component in markdown fixes it, so it's definitely related to markdown parsing

```html
---
title: "Markdown + Tailwind"
setup: |
  import MyButton from '../components/Button.astro';
---

<div class="grid place-items-center h-screen content-center">
    <MyButton>Tailwind Button in Markdown!</MyButton>
    <a href="/" class="p-4 underline">Go home...</a>
</div>
```

**Comment by tony-sull** [comment]
Turns out this wasn't related to the tailwind config, the `latest` branch was out of sync with `main`

A markdown bug was fixed in `1.0-beta.2` but it looks like the the `latest` branch was stuck at `beta.0` so astro.new wasn't pulling in the latest bug fixes
